code/modules/mod/mod_link.dm

```diff
@@ -0,0 +1,407 @@
+	visuals.forceMove(user)
+	animate(visuals, 0.5 SECONDS, alpha = 255)
+	var/datum/callback/setdir_callback = CALLBACK(mod_link.holder, proc_path)
+	setdir_callback.Invoke(user, user.dir, user.dir)
+	mod_link.holder.RegisterSignal(mod_link.holder.loc, COMSIG_ATOM_DIR_CHANGE, proc_path)
+
+/proc/delete_link_visual_generic(datum/mod_link/mod_link)
+	var/mob/living/user = mod_link.get_user_callback.Invoke()
+	playsound(mod_link.get_other().holder, 'sound/machines/terminal_processing.ogg', 50, vary = TRUE, frequency = -1)
+	LAZYREMOVE(mod_link.holder.update_on_z, mod_link.visual)
+	mod_link.holder.lose_hearing_sensitivity(REF(mod_link))
+	mod_link.holder.UnregisterSignal(user, list(COMSIG_CARBON_APPLY_OVERLAY, COMSIG_CARBON_REMOVE_OVERLAY, COMSIG_ATOM_DIR_CHANGE))
+	QDEL_NULL(mod_link.visual)
+
+/proc/on_user_set_dir_generic(datum/mod_link/mod_link, newdir)
+	var/atom/other_visual = mod_link.get_other().visual
+	if(newdir == null)
+		return
+	other_visual.setDir(SOUTH)
+	other_visual.pixel_x = 0
+	other_visual.pixel_y = 0
+	var/matrix/new_transform = matrix()
+	if(newdir & NORTH)
+		other_visual.pixel_y = 13
+		other_visual.layer = BELOW_MOB_LAYER
+		SET_PLANE_IMPLICIT(other_visual, GAME_PLANE_FOV_HIDDEN)
+	if(newdir & SOUTH)
+		other_visual.pixel_y = -24
+		other_visual.layer = ABOVE_ALL_MOB_LAYER
+		SET_PLANE_IMPLICIT(other_visual, GAME_PLANE_UPPER_FOV_HIDDEN)
+		new_transform.Scale(-1, 1)
+		new_transform.Translate(-1, 0)
+	if(newdir & EAST)
+		other_visual.pixel_x = 14
+		other_visual.layer = BELOW_MOB_LAYER
+		SET_PLANE_IMPLICIT(other_visual, GAME_PLANE_FOV_HIDDEN)
+		new_transform.Shear(0.5, 0)
+		new_transform.Scale(0.65, 1)
+	if(newdir & WEST)
+		other_visual.pixel_x = -14
+		other_visual.layer = BELOW_MOB_LAYER
+		SET_PLANE_IMPLICIT(other_visual, GAME_PLANE_FOV_HIDDEN)
+		new_transform.Shear(-0.5, 0)
+		new_transform.Scale(0.65, 1)
+	other_visual.transform = new_transform
+
+/obj/item/mod/control
+	var/datum/mod_link/mod_link
+	var/starting_frequency
+
+/obj/item/mod/control/Initialize(mapload, datum/mod_theme/new_theme, new_skin, obj/item/mod/core/new_core)
+	. = ..()
+	mod_link = new(src, starting_frequency, CALLBACK(src, PROC_REF(get_wearer)), CALLBACK(src, PROC_REF(can_call)), CALLBACK(src, PROC_REF(make_link_visual)), CALLBACK(src, PROC_REF(get_link_visual)), CALLBACK(src, PROC_REF(delete_link_visual)))
+
+/obj/item/mod/control/multitool_act_secondary(mob/living/user, obj/item/multitool/tool)
+	if(!multitool_check_buffer(user, tool))
+		return
+	var/tool_frequency = null
+	if(istype(tool.buffer, /datum/mod_link))
+		var/datum/mod_link/buffer_link = tool.buffer
+		tool_frequency = buffer_link.frequency
+	if(!tool_frequency && mod_link.frequency)
+		tool.set_buffer(mod_link)
+	else if(tool_frequency && !mod_link.frequency)
+		mod_link.frequency = tool_frequency
+	else if(tool_frequency && mod_link.frequency)
+		var/response = tgui_alert(user, "Would you like to copy or imprint the frequency?", "MODLink Frequency", list("Copy", "Imprint"))
```

Comment:
Should put some sanity checks after this alert. Stop people input stalling to set frequency whenever / wherever they want.